### PR TITLE
Notifications: Fixes tableView row selection glitch

### LINKED ISSUE #2975: Notifications SelectedRow not getting Cleared Out
#### Steps:
1. Launch WPiOS
2. Open the Notifications tab
3. Tap over any notification
4. Swipe back quickly

As a result, the selected row will not be cleared out. This happens over 50% of the time.
This bug is affecting Mark 4.8 (and not 4.7), since we've removed *WPTableViewController*.

This appears to be a bug that's [been around since iOS 7](http://stackoverflow.com/questions/19379510/uitableviewcell-doesnt-get-deselected-when-swiping-back-quickly).

**Note:**
*WPTableViewController* had code to manually clear out selection.

### PR DESCRIPTION
**Pull request by jleandroperez**
Fixes #2975

/cc @astralbodies (Thanks!)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m [hunk 1]
```diff
@@ -160,6 +160,9 @@ - (void)viewWillAppear:(BOOL)animated
         _trackedViewDisplay = true;
     }
 
+    // Manually deselect the selected row. This is required due to a bug in iOS7 / iOS8
+    [self.tableView deselectSelectedRowWithAnimation:true];
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: Should this keep to the convention of using `YES` and `NO` for BOOL values in ObjC?  Is your Swift showing ;D

(also on line 160 above)

Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: Huh i didn't notice that!. Thanks Eric, fixing!

**Comment by jleandroperez** [comment]
@aerych Syntax updated (+ merged develop in here!).

May i bother you with a super quick check?.

Thank you sir!

**Comment by aerych** [comment]
I just had the one comment about the bools, so :shipit: sir!

**Comment by jleandroperez** [comment]
Thank you @aerych !

**Pull request merged**